test/auth/oauth_authenticator_spec.js

```diff
@@ -0,0 +1,89 @@
+/* jshint mocha:true */
+var assert = require('assert');
+var rewire = require('rewire');
+var sinon = require('sinon');
+var OauthAuthenticator = rewire('../../lib/auth/oauth_authenticator');
+var BaseBrowserFlow = rewire('../../lib/auth/base_browser_flow');
+var Bluebird = require('bluebird');
+
+describe('OauthAuthenticator', function() {
+
+  describe('#new', function() {
+
+    it('should have default values for options', function() {
+      var oauthAuthenticator = new OauthAuthenticator({});
+      assert.equal(oauthAuthenticator.credentials, null);
+      assert.equal(oauthAuthenticator.flow, null);
+    });
+
+    it('should store access token string as credentials', function() {
+      var oauthAuthenticator = new OauthAuthenticator({
+        credentials: 'token'
+      });
+      assert.deepEqual(oauthAuthenticator.credentials,
+        {'access_token': 'token'});
+      assert.equal(oauthAuthenticator.flow, null);
+    });
+
+    it('should store access token object as credentials', function() {
+      var oauthAuthenticator = new OauthAuthenticator({
+        credentials: {'token_id': 'token_value'}
+      });
+      assert.deepEqual(oauthAuthenticator.credentials,
+        {'token_id': 'token_value'});
+      assert.equal(oauthAuthenticator.flow, null);
+    });
+
+    it('should store a flow', function() {
+      var baseFlow = new BaseBrowserFlow({});
+      var oauthAuthenticator = new OauthAuthenticator({
+        flow: baseFlow
+      });
+      assert.equal(oauthAuthenticator.flow, baseFlow);
+    });
+
+  });
+
+  describe('#authenticateRequest', function() {
+
+    it('should throw an error if credentials are null', function() {
+      var oauthAuthenticator = new OauthAuthenticator({});
+      assert.throws(oauthAuthenticator.authenticateRequest, Error);
+    });
+
+    it('should add auth to request', function() {
+      var oauthAuthenticator = new OauthAuthenticator({
+        credentials: 'token'
+      });
+      var request = oauthAuthenticator.authenticateRequest({});
+      assert.equal(request.headers.Authorization, 'Bearer token');
+    });
+  });
+
+  describe('#ensureCredentials', function() {
+
+    it('should return a resolved Promise with fixed credentials', function(){
+      var oauthAuthenticator = new OauthAuthenticator({
+        credentials: 'token'
+      });
+      var p = oauthAuthenticator.ensureCredentials();
+      assert.equal(true, p.isResolved());
```

Comment:
can we verify that the authenticator stored the credentials? we're not doing that, in spite of the test's name.